Vagrant dev-env support

cpu opened the pull request:
This PR adds support for having a VM based dev env provisioned with Vagrant.

I've selected Ubuntu 14.04 (Trusty) as the base image, but I'm happy to change that if anyone has preferences. I think it's a good common denominator OS. Currently the VM is provisioned with 2GB of RAM

The Go version being installed is fixed at `1.5.3` and there is a hardcoded SHA1 sum in the provisioning script that the release `.tar.gz` is compared against. If there's demand for using `1.6.0` as the default I'm happy to switch this out.

Two bootstrap scripts are run during provisioning:
1. `bootstrap/bootstrap-dev-admin.sh` - invoked as root in the VM
2. `bootstrap/bootstrap-dev.sh` - invoked as `vagrant` user in the VM

The first is primarily used to install Go and set up system wide settings.

The second sets up a Go workspace, adds it to the `vagrant` user's environment, and links the project src from the host machine into the VM's gopath.

One workflow:

1.`vagrant up`
2. modify files in your local git checkout on your host machine
3. `vagrant ssh`
4. `cd workspace/src/github.com/rolandshoemaker/stapled`
5. `go test ./...`
6. etc.

cpu committed:
Adds Github Go .gitignore template.

Adds a copy of [the Github Go
.gitignore](https://github.com/github/gitignore/blob/master/Go.gitignore)
to the project.

cpu committed:
Adds Vagrant support.

This commit adds a Vagrantfile to support having a VM based dev env.

I've selected Ubuntu 14.04 (Trusty) as the base image, but I'm happy to
change that if anyone has preferences. I think it's a good common
denominator OS. Currently the VM is provisioned with 2GB of RAM

The Go version being installed is fixed at 1.5.3 and there is
a hardcoded SHA1 sum in the provisioning script that the release
`.tar.gz` is compared against. If there's demand for using `1.6.0` as
the default I'm happy to switch this out.

Two bootstrap scripts are run during provisioning:

1. `bootstrap/bootstrap-dev-admin.sh` - invoked as root in the VM
2. `bootstrap/bootstrap-dev.sh` - invoked as `vagrant` user in the VM

The first is primarily used to install Go and set up system wide
settings.

The second sets up a Go workspace, adds it to the `vagrant` user's
environment, and links the project src from the host machine into the
VM's gopath.

Workflow:

1.`vagrant up`
2. modify files in your local git checkout on your host machine
3. `vagrant ssh`
4. `cd workspace/src/github.com/rolandshoemaker/stapled`
5. `go test ./...`
6. etc.

cpu commented:
I think (_hope?_) the CI fail is unrelated to this PR.

rolandshoemaker commented:
Travis failure is because the vendored `x/crypto/ocsp` is using a commit that doesn't exist in the upstream repository (because I forgot how to checkout a CL and just applied my patch locally). #18 contains a `.travis.yml` fix that bypasses the default `godep` Travis behavior which I should probably split out to separate PR.

cpu committed:
Changes provisioning to use Go 1.6

Per IRC-chat, better to use Go 1.6 than 1.5.3. This commit changes out
the Go version, the expected SHA1 hash, and removes the explicit vendor
experiment env flag (its on by default in 1.6).

cpu commented:
Using Go 1.6 as of 3f90b1a135dbf510fb688ebc7ce9927bde2af33f
